### DISCUSSION TITLE
CCButton Enable/disable but still read button state?

### DISCUSSION THREAD

**Discussion by fitzgreyve** [tttapa/Control-Surface#1058] [Q&A]
I'm using the enable/dsiable feature on CCButton to control when MIDI output is produced:

```
CCButton AuxDigitalIn {
  3,                                              
  {MIDI_CC::Damper_Pedal, CHANNEL_1},  
}; 

AuxDigitalIn.enable();

AuxDigitalIn.disable();
```
(not posting the full code as it's big)

This works fine.

Ideally, I'd like to be able to still manually detect button transitions while the MIDI output is disabled:

```
if (AuxDigitalIn.getButtonState() == Button::Falling)  { 
```

I realise that the "disable" function removes the CCButton from the list of instances.

Any ideas on how to detect buttin changes with MIDI output disabled ? Define another button in parallel on the same pin, or is there something better?

Many thanks for the library and excellent support!

**Comment by tttapa** [comment]
The Button object inside of the CCButton class is not publicly accessible, so it's probably easiest to use something based on the [Custom-MIDI-Output-Element](https://tttapa.github.io/Control-Surface-doc/Doxygen/d4/d3b/Custom-MIDI-Output-Element_8ino-example.html) example, where you can allow access to the button member variable.

**Comment by fitzgreyve** [comment]
Many thanks, I'm already using a custom CCbutton (sending controller across multiple MIDI channels when "enabled"), so something like:

```
public:
 //Allow the button state to be read directly 
 AH::Button::State getButtonState() const { return button.getButtonState(); }
```
 but I think it needs to be button.update() not button.getButtonState()  if it is to work when the function is "disabled" ?
Not exactly sure of the syntax required here.

**Comment by fitzgreyve** [comment]
think I've got it sorted, typo in the above
```
AH::Button::State getButtonState() const { return button.getButtonState(); }
```
should be:
```
AH::Button::State getButtonState() const { return button.getState(); }
```
so "update" is:

```
 AH::Button::State getButtonState() const { return button.update(); }
```
compiles OK, yet to test it.

**Comment by fitzgreyve** [comment]
Working nicely, thanks for the pointer.
